### PR TITLE
Let gossipd automatically announce the node when a local channel_announcement is queued

### LINKED ISSUE #320: Have gossipd generate node_announcement automatically
1. Change hsmd so gossip can ask for sigs on node_announcement.
2. Pass addresses, color and alias through to gossipd.
3. Have gossipd `handle_channel_announcement` return true if msg is about us, and new.
4. Move signing code from master to gossipd.

### PR DESCRIPTION
**Pull request by cdecker**
This closes #320. I'm spinning this one out of the changes that I'm testing for #284 since the bigger PR has a flaky unit test when running under `valgrind`.

### CODE REVIEW

**Comment by cdecker** [comment]
Ok, added @rustyrussell's changes from #430, and skipped the two that were causing a conflict. I'll fix the merge conflict with `master` tomorrow so we can get this merged :-)

**Comment by cdecker** [comment]
Rebased and should be ready for review. I'll also walk through Rusty's changes and review them.

File: common/daemon_conn.h [hunk 1]
```diff
@@ -43,6 +43,16 @@ void daemon_conn_init(tal_t *ctx, struct daemon_conn *dc, int fd,
 		      struct io_plan *(*daemon_conn_recv)(
 			  struct io_conn *, struct daemon_conn *),
 		      void (*finish)(struct io_conn *, struct daemon_conn *));
+
+/**
+ * daemon_conn_clear - discard a daemon conn without triggering finish.
+ * @dc: the daemon_conn to clean up.
+ *
+ * This is used by gossipd when a peer is handed back, and we no longer
+ * want to deal with it via daemon_conn.  @dc must not be used after this!
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply common/daemon_conn.h]: Hmm, yes, but as you say it's only got one caller, so maybe overkill.

File: lightningd/peer_control.c [hunk 2]
```diff
@@ -183,10 +193,10 @@ void peer_fail_permanent(struct peer *peer, const u8 *msg TAKES)
 	 * zero (ie. all bytes zero), in which case it refers to all
 	 * channels. */
 	static const struct channel_id all_channels;
+	const char *why = tal_strndup(NULL, (char *)msg, tal_len(msg));
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply lightningd/peer_control.c]: Yes, strndup will take() if it needs to, which ahem.. takes care of it. :)

File: tests/test_lightningd.py [hunk 3]
```diff
@@ -2116,6 +2116,61 @@ def test_funding_change(self):
...
+        except ValueError as verr:
+            str(verr).index('to_self_delay {} larger than {}'
+                            .format(max_locktime+1, max_locktime))
+        except Exception as err:
+            self.fail("Unexpected exception {}".format(err))
+        else:
+            self.fail("huge locktime ignored?")
+
+        # We don't have enough left to cover fees if we try to spend it all.
+        try:
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply tests/test_lightningd.py]: Yep, much cleaner. Thanks!

**Comment by rustyrussell** [comment]
Thanks!  Nice work!

**Pull request merged**